spec/deployment_spec.rb

```diff
@@ -114,3 +116,203 @@
+            expect(pod['spec']['containers'][0]['volumeMounts'].count).to be >= 1
+            mount_paths = pod['spec']['containers'][0]['volumeMounts'].map { |it| it['mountPath'] }
+            expect(mount_paths).to include('/var/www')
+          }
+        }
+      end
+
+      it "has a bound persistent volume" do
+        @kubectl.wait_for_deployment(@name)
+
+        wait_until(240,15) {
+          pvcs = @kubectl.get_objects("pvc")['items']
+          expect(pvcs).to_not be_nil
+          expect(pvcs.count).to be >= 1
+
+          pvcs.map! { |pvc| "#{pvc['metadata']['name']}/#{pvc['status']['phase']}" }
+          expect(pvcs).to include("#{@name}/Bound")
+        }
+      end
+
+      if Config.ingress_enabled
+        context 'with an Ingress' do
+          before(:all) do
+            @ingress_filename = 'spec/assets/ingress.yml'
+            @ingress_filename = 'spec/assets/ingress-http.yml' unless Config.lets_encrypt_enabled
+            deploy = @kubectl.deploy(name: @name, filename: @ingress_filename)
+          end
+          after(:all) do
+            delete = @kubectl.delete(name: @name, filename: @ingress_filename)
+
+            ingresses = @kubectl.get_ingresses
+            expect(ingresses).to_not include(@name)
+          end
+
+          if Config.lets_encrypt_enabled
+            context 'with a valid certificate' do
+              before(:all) do
+                wait_until(240,15) {
+                  certificates = @kubectl.get_certificates
+                  expect(certificates).to_not be_nil
+                  expect(certificates.count).to be >= 1
+
+                  expect(certificates.any?{ |c| c['metadata']['name'] == "#{@name}-tls" }).to eq(true)
+                  certificate = certificates.select{ |c| c['metadata']['name'] == "#{@name}-tls" }.first
+
+                  expect(certificate['spec']).to_not be_nil
+                  expect(certificate['spec']['dnsNames']).to_not be_nil
+                  expect(certificate['spec']['dnsNames'].count).to eq(1)
+                  expect(certificate['spec']['dnsNames'][0]).to eq("#{@name}.#{Config.domain}")
+
+                  expect(certificate['status']).to_not be_nil
+                  expect(certificate['status']['conditions']).to_not be_nil
+                  expect(certificate['status']['conditions'].count).to eq(1)
+                  expect(certificate['status']['conditions'][0]['type']).to eq('Ready')
+                  expect(certificate['status']['conditions'][0]['status']).to eq('True')
+
+                  expect(Time.parse(certificate['status']['notAfter']) > (Time.now + 60*60*24*5)).to eq(true)
+                  expect(Time.parse(certificate['status']['notAfter']) < (Time.now + 60*60*24*180)).to eq(true)
+                  expect(Time.parse(certificate['status']['notBefore']) > (Time.now - 60*60*24*1)).to eq(true)
+                }
+              end
+
+              it "can be https queried via domain [#{Config.domain}]" do
```

Comment:
calls  of /hi and /config endpoints can be consolidated into one single test (only /config would be enough), no need for 2 tests.